### PR TITLE
fix: allow blocking IO for setAppLogsPath to avoid DCHECK

### PR DESCRIPTION
**Pull request by loc**
#### Description of Change
Avoid a DCHECK that Slack hits on startup.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: no-notes

### CODE REVIEW

File: shell/browser/api/electron_api_app.cc [hunk 1]
```diff
@@ -846,6 +846,7 @@ void App::SetAppPath(const base::FilePath& app_path) {
 #if !defined(OS_MACOSX)
 void App::SetAppLogsPath(gin_helper::ErrorThrower thrower,
                          base::Optional<base::FilePath> custom_path) {
+  base::ThreadRestrictions::ScopedAllowIO allow_io;
```
Comment (nornagon) [inline shell/browser/api/electron_api_app.cc:849]: Hm, it's unclear to me which bit of this function does IO. Can we wrap this tighter by any chance? Put it closer to where the IO happens?

File: shell/browser/net/proxying_url_loader_factory.cc [hunk 2]
```diff
@@ -810,6 +810,7 @@ void ProxyingURLLoaderFactory::CreateLoaderAndStart(
   network::ResourceRequest request = original_request;
 
   if (ShouldIgnoreConnectionsLimit(request)) {
+    request.priority = net::RequestPriority::MAXIMUM_PRIORITY;
```
Comment (nornagon) [inline shell/browser/net/proxying_url_loader_factory.cc:813]: This seems unrelated to the PR title, is it intentionally here?

File: shell/browser/api/electron_api_app.cc (hunk 1 above)
Comment (loc) [reply shell/browser/api/electron_api_app.cc:849]: base::PathService::Override checks if the directory exists (and thus hits the filesystem). I suppose i can wrap the call in a block and scope it there? but since it happens in both branches, that seemed excessive.

File: shell/browser/net/proxying_url_loader_factory.cc (hunk 2 above)
Comment (loc) [reply shell/browser/net/proxying_url_loader_factory.cc:813]: I can move this into its own PR.

File: shell/browser/api/electron_api_app.cc (hunk 1 above)
Comment (nornagon) [reply shell/browser/api/electron_api_app.cc:849]: It's best to scope this as narrowly as possible IMO, both because it reduces the likelihood we'll accidentally do some IO where we shouldn't be, and because it helps self-document (i.e. i wouldn't have asked this question if it was wrapped in a block with only one call to `PathService::Override`!)

Comment (loc) [reply shell/browser/api/electron_api_app.cc]: Good point! Fixed.
